.gitignore

```diff
@@ -54,6 +54,7 @@ coverage.xml
 *.cover
 .hypothesis/
 .pytest_cache/
+test/
```

Comment:
Curious why we need to ignore this directory.

Reply:
When switching over the test temporary directory generation to use the fixture `tmp_path`, the `cwd` that the test is executed from will create a directory called `test/` in that cwd. Any local run within an IDE of an isolated unit test will create that `test/` directory in a path that git will add. 
It's just to help prevent any stray local tracking server or artifact files from being added to a commit.